make-release.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+# Release process automation script. 
+# Used to create branch/tag, update VERSION files 
+# and and trigger release by force pushing changes to the release branch 
+
+# set to 1 to actually trigger changes in the release branch
+TRIGGER_RELEASE=0 
+NOCOMMIT=0
+
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    '-t'|'--trigger-release') TRIGGER_RELEASE=1; NOCOMMIT=0; shift 0;;
+    '-r'|'--repo') REPO="$2"; shift 1;;
+    '-v'|'--version') VERSION="$2"; shift 1;;
+    '-n'|'--no-commit') NOCOMMIT=1; TRIGGER_RELEASE=0; shift 0;;
+  esac
+  shift 1
+done
+
+usage ()
+{
+  echo "Usage: $0 --repo [GIT REPO TO EDIT] --version [VERSION TO RELEASE] [--trigger-release]"
+  echo "Example: $0 --repo git@github.com:eclipse/che-subproject --version 7.7.0 --trigger-release"; echo
+}
+
+if [[ ! ${VERSION} ]] || [[ ! ${REPO} ]]; then
+  usage
+  exit 1
+fi
+
+# derive branch from version
+BRANCH=${VERSION%.*}.x
+
+# if doing a .0 release, use master; if doing a .z release, use $BRANCH
+if [[ ${VERSION} == *".0" ]]; then
+  BASEBRANCH="master"
+else 
+  BASEBRANCH="${BRANCH}"
+fi
+
+# get sources from ${BASEBRANCH} branch
+echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+git fetch origin "${BASEBRANCH}":"${BASEBRANCH}"
+git checkout "${BASEBRANCH}"
+
+# create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  git branch "${BRANCH}" || git checkout "${BRANCH}" && git pull origin "${BRANCH}"
+  git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}"
+  git checkout "${BRANCH}"
+fi
+
+# change VERSION file
+npm --no-git-tag-version version ${VERSION}
+
+# commit change into branch
+if [[ ${NOCOMMIT} -eq 0 ]]; then
+  COMMIT_MSG="[release] Bump to ${VERSION} in ${BRANCH}"
+  git commit -asm "${COMMIT_MSG}"
+  git pull origin "${BRANCH}"
+  git push origin "${BRANCH}"
+fi
+
+if [[ $TRIGGER_RELEASE -eq 1 ]]; then
+  # push new branch to release branch to trigger CI build
+  git fetch origin "${BRANCH}:${BRANCH}"
+  git checkout "${BRANCH}"
+  docker build -t "quay.io/mkuznets/che-dashboard:${VERSION}" -f apache.Dockerfile .
```

Comment:
It's here just for test, right? Maybe then it's better to make DOCKER_REPO as argument? As it's done for REPO, aka GIT_REPO?

Reply:
this would be a stale constant I think, no need to parametrize it